src/System.Security.Cryptography.Algorithms/tests/RandomNumberGeneratorTests.cs

```diff
@@ -133,6 +133,68 @@ public static void ConcurrentAccess()
             }
         }
 
+#if netstandard17
+        [Fact]
+        public static void GetNonZeroBytes()
+        {
+            using (RandomNumberGenerator rng = RandomNumberGenerator.Create())
+            {
+                Assert.Throws<ArgumentNullException>("data", () => rng.GetNonZeroBytes(null));
+
+                // Array should not have any zeros
+                byte[] rand = new byte[65536];
+                rng.GetNonZeroBytes(rand);
+                Assert.Equal(-1, Array.IndexOf<byte>(rand, 0));
+            }
+        }
+
+        [Fact]
+        public static void GetBytes_Offset()
+        {
+            using (RandomNumberGenerator rng = RandomNumberGenerator.Create())
+            {
+                byte[] rand = new byte[300];
+
+                // Set canary bytes
+                rand[99] = 1;
+                rand[299] = 1;
+
+                rng.GetBytes(rand, 100, 100);
+
+                // Array should not have been touched outside of 100-199
+                Assert.Equal(0, rand[0]);
+                Assert.Equal(99, Array.IndexOf<byte>(rand, 1));
+                Assert.Equal(0, rand[200]);
+                Assert.Equal(299, Array.IndexOf<byte>(rand, 1, 200));
+
+                // Ensure 100-200 has some random bytes; not likely to ever fail here (256^5)
+                Assert.True(rand[100] + rand[125] + rand[150] + rand[175] + rand[199] > 0);
```

Comment:
Something like `rand.Skip(100).Take(100).Sum(b => (int)b)` only false negatives at 256^100.

For best results, make offset and count not be the same, since technically we could be interpreting them backwards.